src/indieweb_utils/parsing/parse.py

```diff
@@ -0,0 +1,40 @@
+import mf2py
+import requests
+from bs4 import BeautifulSoup
+
+
+class RequestError(Exception):
+    pass
+
+
+def _get_soup_from_request(url: str, headers: dict = {}):
```

Comment:
Let's add a return typehint to this function.

🐼 Let's move the "private" functions to the bottom of the file.